sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -424,6 +460,14 @@ RestClient createClient() throws IOException {
             httpAsyncClientBuilder ->
                 httpAsyncClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
       }
+      if (getApiKey() != null) {
+        restClientBuilder.setDefaultHeaders(
+            new Header[] {new BasicHeader("Authorization", "ApiKey " + getApiKey())});
```

Comment:
Is there any way we can test this options easily? If not, never mind.

Reply:
After some research, there is no way in the Elasticsearch SDK to retrieve the headers of a request (or the default headers of the `RestClient`). I don't think there is an "easy way" to test it.